### PR TITLE
Relay refactor 3

### PR DESCRIPTION
**Pull request by dleard**

### CODE REVIEW

File: app/containers/Applications/ApplicationRowItemContainer.js [hunk 1]
```diff
@@ -29,9 +28,18 @@ const ApplicationRowItemContainer = props => {
...
       </td>
       <td>
-        <Button href={applicationDetails} target="_blank" variant="primary">
-          View Application
-        </Button>
+        <Link
+          href={{
+            pathname: '/application-details',
+            query: {
+              applicationId: ciipApplication.rowId,
```
Comment (matthieu-foucault) [inline app/containers/Applications/ApplicationRowItemContainer.js]: Maybe use the relay ID here instead

File: app/lib/relay-environment/index.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// TODO(Dylan): Do we need this import?
```
Comment (matthieu-foucault) [inline app/lib/relay-environment/index.js]: We do not

Comment (matthieu-foucault) [reply app/lib/relay-environment/index.js]: ...unless we do

File: app/pages/_app.js [hunk 3]
```diff
@@ -32,24 +31,24 @@ export default class App extends NextApp {
...
         <QueryRenderer
           environment={environment}
           query={Component.query}
-          variables={variables}
+          variables={{...router.query, ...variables}}
           render={({error, props}) => {
             if (error) return <div>{error.message}</div>;
-            if (props) return <Component {...props} />;
+            if (props)
+              return <Component {...props} router={this.props.router} />;
```
Comment (matthieu-foucault) [inline app/pages/_app.js]:
```suggestion
              return <Component {...props} />;
```

File: schema/revert/table_application.sql [hunk 4]
```diff
@@ -0,0 +1,7 @@
+-- Revert ggircs-portal:table_application from pg
+
+BEGIN;
```
Comment (matthieu-foucault) [inline schema/revert/table_application.sql]:
```suggestion
begin;
```

File: schema/revert/table_application.sql [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+
+drop table ggircs_portal.application;
+
+COMMIT;
```
Comment (matthieu-foucault) [inline schema/revert/table_application.sql]:
```suggestion
commit;
```

File: schema/sqitch.plan [hunk 6]
```diff
@@ -4,12 +4,13 @@
...
-table_application_status [schema_ggircs_portal function_update_timestamps] 2019-08-29T19:42:02Z Dylan Leard <dylan@button.is> # add table application_status to schema ggircs_portal
+table_user [schema_ggircs_portal] 2019-07-11T18:32:05Z Hamza Javed <hamza@button.is> # Create a User table for GGIRCS Industry Portal (GIP)
+table_product [schema_ggircs_portal] 2019-08-06T20:15:37Z Hamza Javed <hamza@button.is> # A table for adding products (for benchmarks)
+table_benchmark [table_product] 2019-08-06T20:25:30Z Hamza Javed <hamza@button.is> # A table for Benchmarks and Eligibility Thresholds for products
+table_application [schema_ggircs_portal] 2019-10-11T18:09:42Z Dylan Leard <dylan@button.is> # Application table
+table_form_json [schema_ggircs_portal] 2019-07-24T20:00:52Z Hamza Javed <hamza@button.is> # Table for the survey js forms (in JSON)
+table_form_result [schema_ggircs_portal table_form_json table_user] 2019-07-24T22:44:27Z Hamza Javed <hamza@button.is> # A place to store all the form results
+table_application_status [schema_ggircs_portal] 2019-08-29T19:42:02Z Dylan Leard <dylan@button.is> # add table application_status to schema ggircs_portal
```
Comment (matthieu-foucault) [inline schema/sqitch.plan]: re-add the dependencies that were deleted

File: schema/verify/table_application.sql [hunk 7]
```diff
@@ -0,0 +1,7 @@
+-- Verify ggircs-portal:table_application on pg
+
+BEGIN;
```
Comment (matthieu-foucault) [inline schema/verify/table_application.sql]:
```suggestion
begin;
```

File: schema/verify/table_application.sql [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+
+select pg_catalog.has_table_privilege('ggircs_portal.application', 'select');
+
+ROLLBACK;
```
Comment (matthieu-foucault) [inline schema/verify/table_application.sql]:
```suggestion
rollback;
```

File: app/pages/_app.js (hunk 3 above)
Comment (dleard) [reply app/pages/_app.js]: This is needed. I think the props being spread into the component are different props than `this.props`

**Pull request merged**